### PR TITLE
Implementing stream commands

### PR DESCRIPTION
**Pull request by pratheep-kumar**
# Stream Commands Implementation

This PR adds full support for Valkey/Redis Stream commands to `Valkey::Commands::StreamCommands`. Each command is implemented following the same pattern as other command modules, with comprehensive test coverage, 100% compatibility with `redis-rb`, and alignment with other Glide language bindings.

**Reference**: [Valkey Stream Commands Documentation](https://valkey.io/commands/#stream)

### Core Stream Commands

| Redis CLI Command | Ruby Method | Description |
|-------------------|-------------|-------------|
| `XADD` | `xadd(key, entry, approximate: nil, maxlen: nil, minid: nil, nomkstream: nil, id: "*")` | Append a new entry to a stream |
| `XDEL` | `xdel(key, *ids)` | Remove one or more entries from a stream |
| `XLEN` | `xlen(key)` | Get the length of a stream |
| `XRANGE` | `xrange(key, start, end_id, **options)` | Get entries from a stream within a range of IDs |
| `XREVRANGE` | `xrevrange(key, end_id = "+", start = "-", count: nil)` | Get entries from a stream in reverse order |
| `XREAD` | `xread(keys, ids, count: nil, block: nil)` | Read entries from one or more streams |
| `XTRIM` | `xtrim(key, maxlen, strategy: "MAXLEN", approximate: true)` | Trim a stream to a maximum length |

### Consumer Group Commands

| Redis CLI Command | Ruby Method | Description |
|-------------------|-------------|-------------|
| `XGROUP CREATE` | `xgroup_create(key, group, id, **options)` | Create a consumer group for a stream |
| `XGROUP CREATECONSUMER` | `xgroup_createconsumer(key, group, consumer)` | Create a consumer in a consumer group |
| `XGROUP SETID` | `xgroup_setid(key, group, id)` | Set the last-delivered ID for a consumer group |
| `XGROUP DESTROY` | `xgroup_destroy(key, group)` | Destroy a consumer group |
| `XGROUP DELCONSUMER` | `xgroup_delconsumer(key, group, consumer)` | Remove a consumer from a consumer group |
| `XGROUP` | `xgroup(subcommand, key, group, *args, **options)` | Dispatcher method for all XGROUP subcommands |

### Consumer Group Reading Commands

| Redis CLI Command | Ruby Method | Description |
|-------------------|-------------|-------------|
| `XREADGROUP` | `xreadgroup(group, consumer, keys, ids, count: nil, block: nil, noack: false)` | Read entries from streams using a consumer group |
| `XACK` | `xack(key, group, *ids)` | Acknowledge one or more messages in a consumer group |
| `XPENDING` | `xpending(key, group, *args, idle: nil)` | Get information about pending messages in a consumer group |
| `XCLAIM` | `xclaim(key, group, consumer, min_idle_time, ids, **options)` | Claim ownership of pending messages |
| `XAUTOCLAIM` | `xautoclaim(key, group, consumer, min_idle_time, start, **options)` | Automatically claim pending messages |

### Stream Information Commands

| Redis CLI Command | Ruby Method | Description |
|-------------------|-------------|-------------|
| `XINFO STREAM` | `xinfo_stream(key, **options)` | Get information about a stream |
| `XINFO GROUPS` | `xinfo_groups(key)` | Get information about consumer groups |
| `XINFO CONSUMERS` | `xinfo_consumers(key, group)` | Get information about consumers in a group |
| `XINFO` | `xinfo(subcommand, key, group = nil, **options)` | Dispatcher method for all XINFO subcommands |

**Total: 20 methods implemented (7 core + 6 consumer group + 4 reading + 3 info)**

**Note**: `XSETID` command is not yet implemented as it requires backend support in glide-core (request type 1420).

### 100% redis-rb Compatibility

- **API Parity**: All methods match `redis-rb` API exactly, including parameter order and default values
- **Response Format**: Returns flat arrays `[["id", ["field", "value", ...]], ...]` matching `redis-rb` format exactly
- **Return Types**: All return values match `redis-rb` behavior precisely
- **Argument Handling**: Variadic arguments, optional parameters, and keyword arguments work identically
- **Method Signatures**: Parameter order matches `redis-rb` (e.g., `xadd` has `id` as last parameter, `xrevrange` parameter order)

### Stream Response Format

- **Flat Arrays**: All stream entry responses return `[["id", ["field", "value", ...]], ...]` format (redis-rb compatible)
- **Hash Conversion**: Stream responses in `xread`/`xreadgroup` return `{"stream" => [[id, [fields...]], ...]}` format
- **Consistent Formatting**: `xrange`, `xrevrange`, `xclaim`, and `xautoclaim` all use the same entry format
- **Proper Handling**: Empty streams, nil responses, and edge cases are handled correctly

### Consumer Group Support

- **Full Implementation**: All consumer group management commands (create, destroy, setid, etc.)
- **Consumer Management**: Create and delete consumers within groups
- **Pending Messages**: Full support for pending message inspection and management
- **Message Claiming**: Both manual (`xclaim`) and automatic (`xautoclaim`) message claiming
- **Acknowledgment**: Support for acknowledging processed messages

### Advanced Features

- **Blocking Reads**: Support for blocking `xread` and `xreadgroup` operations
- **Stream Trimming**: Support for both approximate and exact trimming with `MAXLEN` strategy
- **Auto-Generated IDs**: Support for auto-generating stream entry IDs
- **Custom IDs**: Support for specifying custom entry IDs
- **Stream Info**: Comprehensive stream, group, and consumer information retrieval

### Production Ready

- **Comprehensive Tests**: 20+ test cases covering all commands and edge cases
- **Error Handling**: Proper error handling for invalid operations and edge cases
- **Documentation**: Complete RDoc documentation with examples for all methods
- **Version Checks**: Tests skip gracefully when features not available (e.g., `xautoclaim` requires Redis 6.2+)
- **Cleanup**: Tests properly clean up created streams and groups
- **Rubocop Compliant**: All code passes rubocop linting

### Test Coverage

✅ **Lint Tests** (`test/lint/stream_commands.rb`)
- 20+ comprehensive test cases
- Tests all command categories (core, consumer groups, reading, info)
- Version detection for Redis 6.2+ features (`xautoclaim`, `xgroup_createconsumer`)
- Proper setup/teardown with cleanup
- Edge case testing (empty streams, nil responses, etc.)

### Test Categories

- ✅ **Core Commands**: `xadd`, `xdel`, `xlen`, `xrange`, `xrevrange`, `xread`, `xtrim`
- ✅ **Consumer Groups**: `xgroup_create`, `xgroup_destroy`, `xgroup_setid`, `xgroup_createconsumer`, `xgroup_delconsumer`
- ✅ **Reading**: `xreadgroup`, `xack`, `xpending`, `xclaim`, `xautoclaim`
- ✅ **Information**: `xinfo_stream`, `xinfo_groups`, `xinfo_consumers`

### Implementation Files

- **`lib/valkey/commands/stream_commands.rb`**: Complete implementation of all 20 stream commands
- **`lib/valkey/utils.rb`**: Added utility functions for stream response conversion:
  - `HashifyStreamEntries`: Converts stream entries to flat array format (redis-rb compatible)
  - `HashifyStreams`: Converts stream responses for `xread`/`xreadgroup`
  - `HashifyStreamAutoclaim`: Converts `xautoclaim` responses
  - `HashifyStreamPendings`: Converts `xpending` summary responses
  - `HashifyStreamPendingDetails`: Converts `xpending` detailed responses

### Test Files

- **`test/lint/stream_commands.rb`**: Comprehensive test suite with 20+ test cases

### Key Implementation Details

1. **Response Format Matching**: All responses match `redis-rb` format exactly (flat arrays, not hashes)
2. **Parameter Order**: Method signatures match `redis-rb` exactly (e.g., `xadd` has `id` as last parameter)
3. **Boolean to Integer Conversion**: `xgroup_destroy` and `xgroup_createconsumer` convert boolean responses to integers
4. **Nil Handling**: Proper handling of nil responses for `xread`, `xreadgroup`, and empty streams
5. **Edge Cases**: Comprehensive handling of empty arrays, nil values, and malformed responses

## 📝 Notes

- **XSETID Not Implemented**: The `xsetid` command is commented out as it requires backend support (request type 1420). Once enabled in glide-core, it can be uncommented and tested.
- **Redis Version Requirements**: Some commands require specific Redis versions:
  - `xautoclaim`: Redis 6.2+
  - `xgroup_createconsumer`: Redis 6.2+
  - All other commands: Redis 5.0+

## 🎯 Compatibility

- ✅ **redis-rb**: 100% API compatibility
- ✅ **Valkey**: Full support for all stream commands
- ✅ **Redis**: Full support for Redis 5.0+ stream commands
- ✅ **Glide Backend**: Uses all available stream command request types

## 📚 Examples

```ruby
# Add entries to a stream
id = valkey.xadd("mystream", { "field1" => "value1", "field2" => "value2" })

# Read entries
entries = valkey.xrange("mystream", "-", "+")
# => [["1234567890-0", ["field1", "value1", "field2", "value2"]], ...]

# Create consumer group
valkey.xgroup_create("mystream", "mygroup", "0")

# Read from consumer group
result = valkey.xreadgroup("mygroup", "consumer1", ["mystream"], [">"])
# => { "mystream" => [["1234567890-0", ["field1", "value1", ...]], ...] }

# Acknowledge messages
valkey.xack("mystream", "mygroup", id)

# Get stream info
info = valkey.xinfo_stream("mystream")
```

## ✅ Checklist

- [x] All 20 stream commands implemented
- [x] 100% redis-rb API compatibility
- [x] Comprehensive test coverage (20+ test cases)
- [x] All tests passing
- [x] Rubocop compliant
- [x] Complete RDoc documentation
- [x] Proper error handling
- [x] Edge case handling
- [x] Version detection for Redis 6.2+ features

### CODE REVIEW

File: test/lint/stream_commands.rb [hunk 1]
```diff
@@ -0,0 +1,464 @@
+# frozen_string_literal: true
+
+module Lint
+  module StreamCommands
+    def test_xadd
+      target_version "5.0" do
```
Comment (Sasidharan3094) [inline test/lint/stream_commands.rb]: As we discussed earlier, please remove the target_version. Let's add this for the ones which are introduced from valkey 8.
